libs/ts-rest/fastify/src/lib/ts-rest-fastify.ts

```diff
@@ -127,7 +157,19 @@ export const initServer = () => ({
       requestValidationErrorHandler: 'combined',
     }
   ) => {
-    recursivelyRegisterRouter(routerImpl.routes, contract, [], app, options);
+    const { hooks = {}, ...restOfOptions } = options;
+    recursivelyRegisterRouter(
+      routerImpl.routes,
+      contract,
+      [],
+      app,
+      restOfOptions
+    );
+
+    Object.keys(hooks).forEach((hookName) =>
+      // @ts-expect-error Fastify's hook overload seems to be complaining here for no reason afaik
+      app.addHook(hookName, hooks[hookName])
+    );
```

Comment:
This will add the hooks to the entire app rather than just the routes of the passed contract. We should forward the entire `options` object to `recursivelyRegisterRouter` and add these "global" hooks to each route individually right before the route's own hooks.